tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/UserEvents/HttpContextSetUser.cs

```diff
@@ -0,0 +1,116 @@
+// <copyright file="HttpContextSetUser.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+#if !NETFRAMEWORK
+using System;
+using System.ComponentModel;
+using System.Security.Claims;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Coordinator;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore.UserEvents
+{
+    /// <summary>
+    /// The ASP.NET Core middleware integration.
+    /// public static Task SignInAsync(this HttpContext context, string? scheme, ClaimsPrincipal principal, AuthenticationProperties? properties)
+    /// if we make it till here, it means sign in has succeeded
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = AssemblyName,
+        TypeName = HttpContextExtensionsTypeName,
+        ParameterTypeNames = ["System.Security.Claims.ClaimsPrincipal"],
+        MethodName = "set_User",
+        ReturnTypeName = ClrNames.Task,
+        MinimumVersion = Major2,
+        MaximumVersion = SupportedVersions.LatestDotNet,
+        IntegrationName = nameof(IntegrationId.AspNetCore),
+        InstrumentationCategory = InstrumentationCategory.AppSec)]
+    [InstrumentMethod(
+        AssemblyName = AssemblyName,
+        TypeName = HttpContextExtensionsTypeName,
+        ParameterTypeNames = ["System.Security.Claims.ClaimsPrincipal"],
+        MethodName = "set_User",
+        ReturnTypeName = ClrNames.Task,
+        MinimumVersion = Major2,
+        CallTargetIntegrationKind = CallTargetKind.Derived,
```

Comment:
Derived integration are much more expensive in general  - IIRC, we have to analyze the class hierarchy for every loaded type to see if it derives from the type we're interested in. What I don't know is if the incremental perf hit is significant, or if it's just "as soon as we have a derived instrumentation we pay the cost".

I wonder if it would be sufficient to just instrument the "known" public `HttpContext` implementations e.g. `DefaultHttpContext` - realistically no-one outside of Microsoft is implementing their own `HttpContext` is going to be implementing their own HttpContext, so will probably be good enough? 🤔

Reply:
yes good point thanks, changed in [bf30476](https://github.com/DataDog/dd-trace-dotnet/pull/6431/commits/bf30476c7b3d6c810cc4d5a9647ae2e30aa41099) and [f29295d](https://github.com/DataDog/dd-trace-dotnet/pull/6431/commits/f29295d64625775f3c3ce6fe7f473a964cf9bb00)